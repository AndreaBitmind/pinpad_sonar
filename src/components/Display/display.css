.display {
  display: flex;
  border: 1px solid black;
  width: 100%;
  min-height: 55px;
  padding: 5px 0;
  border-radius: 15px;
  text-align: center;
  align-items: center;
  justify-content: center;
  font-size: 2rem;
  background-color: rgba(128, 128, 128, 0.327);
}
