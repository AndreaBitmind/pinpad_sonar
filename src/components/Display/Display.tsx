import "./display.css";

interface DisplayProps {
  result: string[];
}

const Display = ({ result }: DisplayProps): JSX.Element => {
  return <div className="display">{result}</div>;
};

export default Display;
