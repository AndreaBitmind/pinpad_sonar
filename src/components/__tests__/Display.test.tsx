import TestRenderer from "react-test-renderer";
import Display from "../Display/Display";

describe("Given a Display component", () => {
  describe("When it's rendered in the screen", () => {
    test("Then it should show the Display component", () => {
      const resultTest = ["*", "*", "*", "5"];

      const expectedDisplayComponent = TestRenderer.create(
        <Display result={resultTest} />
      );

      expect(expectedDisplayComponent).toMatchSnapshot();
    });
  });
});
