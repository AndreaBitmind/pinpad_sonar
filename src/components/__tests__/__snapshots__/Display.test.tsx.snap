// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Display component When it's rendered in the screen Then it should show the Display component 1`] = `
<div
  className="display"
>
  *
  *
  *
  5
</div>
`;
